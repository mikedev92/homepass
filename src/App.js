// @flow
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { addNavigationHelpers } from 'react-navigation';
import { ApolloClient } from 'apollo-client';
import { withApollo } from 'react-apollo';
import { compose } from 'recompose';
import Navigator from './components/Navigator';
import { navigationSelector, type State } from './reducers';

type Props = {
	dispatch: Dispatch,
	state: State,
	client: ApolloClient
};

class App extends Component {
	props: Props;
	interval: number;

	render() {
		const { dispatch, state } = this.props;
		return (
			<Navigator
				navigation={addNavigationHelpers({
					dispatch,
					state
				})}
			/>
		);
	}
}

export default compose(withApollo, connect(state => ({ state: navigationSelector(state) })))(App);
