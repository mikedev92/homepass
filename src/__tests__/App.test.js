import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';

describe('<App />', () => {
  it('renders a navigator', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.contains('Navigator'));
    expect(wrapper).toMatchSnapshot();
  });
});

