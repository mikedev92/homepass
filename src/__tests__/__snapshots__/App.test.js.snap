// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders a navigator 1`] = `<Stub />`;
